internal/controllers/topology/cluster/patches/external/external_patch_generator.go

```diff
@@ -48,6 +48,11 @@ func (e externalPatchGenerator) Generate(ctx context.Context, forObject client.O
 	if !feature.Gates.Enabled(feature.RuntimeSDK) {
 		return nil, errors.Errorf("can not use external patch %q if RuntimeSDK feature flag is disabled", *e.patch.External.GenerateExtension)
 	}
+
+	// Set the settings defined in external patch definition on the request object.
+	// These settings will override overlapping keys defined in ExtensionConfig settings.
+	req.Settings = e.patch.External.Settings
```

Comment:
Given that we are re-using req for all patches. I think we should set req.Settings to nil after the call (maybe defer to capture all cases?)

Reply:
Good catch! 
The next call to the generator should reset the settings to the value of that patch call but it is safer to reset it with in the same call.